src/d3a/models/strategy/external_strategies/storage.py

```diff
@@ -0,0 +1,261 @@
+            self.offers.remove_by_id(arguments["offer"])
+            self.redis.publish_json(response_channel,
+                                    {"status": "ready", "deleted_offer": arguments["offer"]})
+        except Exception as e:
+            logging.error(f"Error when handling offer delete on area {self.device.name}: "
+                          f"Exception: {str(e)}, Offer Arguments: {arguments}")
+            self.redis.publish_json(
+                response_channel,
+                {"status": "error",
+                 "error_message": f"Error when handling offer delete "
+                                  f"on area {self.device.name} with arguments {arguments}."})
+
+    def _offer(self, payload):
+        offer_response_channel = f'{self.device.name}/offer/response'
+        if not check_for_connected_and_reply(self.redis, offer_response_channel,
+                                             self.connected):
+            return
+        try:
+            arguments = json.loads(payload["data"])
+            assert set(arguments.keys()) == {'price', 'energy'}
+            arguments['seller'] = self.device.name
+            arguments['seller_origin'] = self.device.name
+        except Exception as e:
+            logging.error(f"Incorrect offer request. Payload {payload}. Exception {str(e)}.")
+            self.redis.publish_json(
+                offer_response_channel,
+                {"error": "Incorrect offer request. Available parameters: (price, energy)."}
+            )
+        else:
+            self.pending_requests.append(
+                IncomingRequest("offer", arguments, offer_response_channel))
+
+    def _offer_impl(self, arguments, response_channel):
+        try:
+            energy_sell_dict = self.state.clamp_energy_to_sell_kWh([self.market.time_slot])
+            assert energy_sell_dict[self.market.time_slot] >= arguments['energy']
+            expected_energy = self.state.used_storage * (1 - self.state.min_allowed_soc_ratio)
+            assert arguments['energy'] <= expected_energy
+            offer = self.market.offer(**arguments)
+            # self.state.offered_sell_kWh[self.market.time_slot] += offer.energy
+
+            self.offers.post(offer, self.market.id)
+            self.state.offered_sell_kWh[self.market.time_slot] += offer.energy
+            self.redis.publish_json(response_channel,
+                                    {"status": "ready", "offer": offer.to_JSON_string()})
+        except Exception as e:
+            logging.error(f"Error when handling offer create on area {self.device.name}: "
+                          f"Exception: {str(e)}, Offer Arguments: {arguments}")
+            self.redis.publish_json(
+                response_channel,
+                {"status": "error",
+                 "error_message": f"Error when handling offer create "
+                                  f"on area {self.device.name} with arguments {arguments}."})
+
+    def _list_bids(self, payload):
```

Comment:
I wonder whether we add the list_offers and list_bids also to the ExternalMixin as they are used on multiple occasions (i know, a load should not be able to call list_offers, but we could disable this by checking for the type. 
What you you think. Also tagging @spyrostz

Reply:
If you continue this point, functions like _delete_offer, _offer etc, too actually.

Reply:
I would recommend having 2 different mixins, one for the bids and one for the offers, if we plan to go this way. If 2 separate mixins are used, then the load can use only the bid-related one, the pv the offer-related one, and storage both. However, if it feels like an overkill feel free to skip it, it does not look that important to my eyes

Reply:
Please skip the refactoring then. Thanks!